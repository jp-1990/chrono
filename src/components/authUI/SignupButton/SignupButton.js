import React from "react";
import { View, Text, Pressable, StyleSheet } from "react-native";

const SignupButton = ({ labelColor, buttonColor, ripple, onPress }) => {
  return (
    <View style={styles.signupBox}>
      <Text style={{ ...styles.signupText, color: labelColor || "#454851" }}>
        Don't have an account?
      </Text>
      <Pressable
        android_ripple={{
          color: ripple || "#B5D2FD",
          borderless: false,
          radius: 150,
        }}
        style={styles.signupPressBox}
        onPress={() => onPress()}
      >
        <Text
          style={{ ...styles.signupPress, color: buttonColor || "#5484CB" }}
        >
          Sign up
        </Text>
      </Pressable>
    </View>
  );
};

export default SignupButton;

const styles = StyleSheet.create({
  signupBox: { justifyContent: "center", alignItems: "center", marginTop: 72 },
  signupText: {
    fontSize: 18,
  },
  signupPressBox: {
    marginVertical: 6,
  },
  signupPress: {
    fontSize: 18,
    fontFamily: "lato-bold",
    paddingHorizontal: 18,
    paddingVertical: 6,
  },
});
