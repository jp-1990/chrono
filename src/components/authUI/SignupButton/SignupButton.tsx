import React from "react";
import { View, Pressable, StyleSheet } from "react-native";

import { Text } from "../../Common";

interface Props {
  labelColor?: string;
  buttonColor?: string;
  ripple?: string;
  onPress(): void;
}

const SignupButton: React.FC<Props> = ({
  labelColor,
  buttonColor,
  ripple,
  onPress,
}) => {
  return (
    <View style={styles.signupBox}>
      <Text variant="h4" style={{ color: labelColor || "#454851" }}>
        Don't have an account?
      </Text>
      <Pressable
        android_ripple={{
          color: ripple || "#B5D2FD",
          borderless: false,
          radius: 150,
        }}
        style={styles.signupPressBox}
        onPress={() => onPress()}
      >
        <Text
          variant="h4"
          style={{ ...styles.signupPress, color: buttonColor || "#5484CB" }}
        >
          Sign up
        </Text>
      </Pressable>
    </View>
  );
};

export default SignupButton;

const styles = StyleSheet.create({
  signupBox: {
    justifyContent: "center",
    alignItems: "center",
    marginTop: 56,
    marginBottom: 18,
  },
  signupPressBox: {
    marginVertical: 4,
  },
  signupPress: {
    fontFamily: "lato-bold",
    paddingHorizontal: 18,
    paddingVertical: 6,
  },
});
