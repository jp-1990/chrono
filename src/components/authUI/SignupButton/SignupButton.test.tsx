import React from "react";
import TestRenderer from "react-test-renderer";
import { render, cleanup, fireEvent } from "@testing-library/react-native";

import SignupButton from "./SignupButton";

describe("<SignupButton />", () => {
  afterEach(cleanup);

  const mockCallback = jest.fn();
  const testProps = { onPress: mockCallback };
  const testRender = TestRenderer.create(<SignupButton {...testProps} />);

  it("renders correctly", () => {
    expect(testRender).toMatchSnapshot();
  });

  it("uses the onPress prop when it is pressed", () => {
    const { getByText } = render(<SignupButton {...testProps} />);
    fireEvent.press(getByText("Sign up"));
    expect(mockCallback.mock.calls.length).toEqual(1);
    mockCallback.mockClear();
  });

  it("does not use the onPress prop when it is not pressed", () => {
    const { getByText } = render(<SignupButton {...testProps} />);
    expect(getByText("Sign up")).toBeTruthy();
    expect(mockCallback.mock.calls.length).toEqual(0);
    mockCallback.mockClear();
  });
});
