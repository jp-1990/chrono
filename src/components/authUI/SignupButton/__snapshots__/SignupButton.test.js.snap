// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SignupButton /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
      "marginBottom": 36,
      "marginTop": 72,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#454851",
        "fontSize": 18,
      }
    }
  >
    Don't have an account?
  </Text>
  <View
    accessible={true}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "marginVertical": 6,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#5484CB",
          "fontFamily": "lato-bold",
          "fontSize": 18,
          "paddingHorizontal": 18,
          "paddingVertical": 6,
        }
      }
    >
      Sign up
    </Text>
  </View>
</View>
`;
